DataFlowDiagrams/DesigningForSSE.md

```diff
@@ -9,3 +9,57 @@
+
+From our analysis of the automatically generated threats, it seems our biggest gaps in threat mitigation exist in tampering and denial of service.  Tampering threats exist in the API request from the bank employee to the Elasticsearch primary node, the index request from the Elasticsearch primary node to the file system, the index response from the file system to the Elasticsearch primary node, the replica index request from the Elasticsearch replica node to the replica file system, the replica index response from the replica file system to the Elasticsearch replica node, the replication request from the Elasticsearch primary node to the Elasticsearch replica node, and in the replication response from the Elasticsearch replica node to the Elasticsearch primary node.  Since this exists in 7 out of 8 data flows, the design seems particularly vulnerable to tampering threats.  The denial of service threats exist in the API request from the bank employee to the Elasticsearch primary node, exist in the API response from the Elasticsearch primary node to the bank employee, the index request from the Elasticsearch primary node to the file system, the replica index request from the Elasticsearch replica node to the replica file system, and the replication request from the Elasticsearch primary node to the Elasticsearch replica node.  The denial of service threat exists in 5 out of 8 data flows.  As a result, it is reasonable to conclude the design is also particularly vulnerable to denial of service threats.
+
+Elasticsearch threats that are handled well are encrypting communications and authenticating clients to the cluster.  Elasticsearch can be configured to use TLS/SSL for communication between nodes in a cluster and between a client and the cluster. This mitigated several threats related to tampering.  Elasticsearch can be configured to authenticate clients which mitigates several spoofing threats.
+
+Gaps in Elasticsearch security from DFD are cluster/node security, securing log files, changing Elasticsearch execution flow via API requests, and handling issues with replication in a cluster.  ES adds nodes to a cluster using a config file with a list of IP addresses where the other nodes live (https://www.elastic.co/guide/en/elasticsearch/reference/current/important-settings.html#unicast.hosts). There is then a module that handles discovering other nodes in the cluster using this list and its own cluster name setting. If this list were compromised on one node the whole cluster could be compromised by a potentially malicious node being added.  Could not find any documentation on how ES ensures the integrity of its log files.  Unsure if there are ways to send in API requests to an ES cluster that change code execution flow in such a way to elevate the privilege of a user.  Could not find any documentation around what happens when a primary node tries and fails to communicate with a replica node in order to propagate a data change to a replica shard.
```

Comment:
We have some sentences here that start with 'Unsure.., Could not...', we should write full sentences: 'We could not...', 'We are unsure...'